### PR TITLE
Scoreboard: Fixed scoreboard and removed obsolete functions

### PR DESCRIPTION
**Pull request by Alf21**
Finally, we got rid of PlayerScore

### CODE REVIEW

**Comment by Alf21** [comment]
PR has to wait for <https://github.com/Facepunch/sbox-issues/issues/1032>

File: code/gamemode/Game.cs [hunk 1]
```diff
@@ -115,9 +116,19 @@ public override void ClientDisconnect(Client client, NetworkDisconnectionReason
...
 
+            Event.Run(TTTEvent.Player.Disconnected, client.SteamId, reason);
+
+            ClientClientDisconnect(client.SteamId, reason);
+
             base.ClientDisconnect(client, reason);
         }
 
+        [ClientRpc]
+        public static void ClientClientDisconnect(ulong steamId, NetworkDisconnectionReason reason)
```
Comment (mzegar) [inline code/gamemode/Game.cs:127]: `ClientClient` here

File: code/player/TTTPlayer.Commands.Server.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public static void RespawnPlayer()
 
             TTTPlayer player = ConsoleSystem.Caller.Pawn as TTTPlayer;
 
-            if (!player.IsValid() || ConsoleSystem.Caller.GetScore<bool>("forcedspectator", false))
+            if (!player.IsValid() || ConsoleSystem.Caller.GetValue<bool>("forcedspectator", false))
```
Comment (mzegar) [inline code/player/TTTPlayer.Commands.Server.cs:25]: I wonder if we should hardcore these strings somewhere. `forcedspectator`

**Review by mzegar — COMMENTED** [review]
I just did a quick review on my phone. You could also use `UserId` instead of `SteamId` but both are unique anyways.

File: code/gamemode/Game.cs (hunk 1 above)
Comment (Alf21) [reply code/gamemode/Game.cs:127]: Yea, sadly, Intended :D

File: code/player/TTTPlayer.Commands.Server.cs (hunk 2 above)
Comment (Alf21) [reply code/player/TTTPlayer.Commands.Server.cs:25]: Like in the event system? Could do so if we plan introducing more of these, currently, "forcedspectator" is the only one.

**Comment by Alf21** [comment]
> I just did a quick review on my phone. You could also use `UserId` instead of `SteamId` but both are unique anyways.

UserId and NetworkIdent are reused Ids, steamid working as intended (if a user reconnects and the list didn't got cleaned up, that's no problem with steamid)

**Comment by mzegar** [comment]
> > I just did a quick review on my phone. You could also use `UserId` instead of `SteamId` but both are unique anyways.
> 
> UserId and NetworkIdent are reused Ids, steamid working as intended (if a user reconnects and the list didn't got cleaned up, that's no problem with steamid)

`User ID of this client. These IDs will never be reused, even for the same player after rejoin.`

**Review by mzegar — APPROVED** [review]
LGTM

**Comment by mzegar** [comment]
Merging since game mode is completely busted without this. Thanks.

**Pull request merged**